File: lua/scrabble-score/scrabble-score.lua
SCORE = {
	a = 1,
	e = 1,
	i = 1,
	o = 1,
	u = 1,
	l = 1,
	n = 1,
	r = 1,
	s = 1,
	t = 1,
	d = 2,
	g = 2,
	b = 3,
	c = 3,
	m = 3,
	p = 3,
	f = 4,
	h = 4,
	v = 4,
	w = 4,
	y = 4,
	k = 5,
	j = 8,
	x = 8,
	q = 10,
	z = 10,
}
local function score(word)
	if word == nil then
		return 0
	end

	word = string.lower(word)
	local total = 0
	for i = 1, #word do
		total = total + SCORE[string.sub(word, i, i)]
	end
	return total
end

return { score = score }

File: dart/scrabble-score/lib/scrabble_score.dart
Map<String, int> table = const {
  'A, E, I, O, U, L, N, R, S, T': 1,
  'D, G': 2,
  'B, C, M, P': 3,
  'F, H, V, W, Y': 4,
  'K': 5,
  'J, X': 8,
  'Q, Z': 10,
};

int score(String word) {
  Map<String, int> scores = {};
  table.forEach(
    (letters, score) =>
        letters.split(', ').forEach((letter) => scores[letter] = score),
  );

  return word
      .toUpperCase()
      .split("")
      .fold(0, (acc, i) => acc + (scores[i] ?? 0));
}
